### PR TITLE
Javier cueto final delivery

### PR DESCRIPTION
**Pull request by javierCueto**
#### What does this PR do?
- improving network layer using combine
- const with localization
- Spanish support
- Fixes in search module 
- Show error from api
- Fixes interface supporting ipad device
- implementing workers for modules (moviesWorker, reviewWorker ...)
- updating test

#### Where should the reviewer start?
- modules view
- network layer

#### Code review checklist [Code Review Rubric](https://github.com/wizeline/iOS-apprenticeship-program-2020/blob/main/code-review-rubric.md)

1. DRY Principle (Don’t repeat yourself)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

2. Yagni Principle (You Aren't Gonna Need It)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

3. KISS Principle (Keep It Simple, St...d)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

4. Remove unused code

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

5. Hard-coded values

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

6. Language style guide & coding conventions
 
 - [Wizeline Style Guide](https://github.com/wizeline/mobile-documentation/blob/master/guidelines/swift-style-guide.md) 
 - [RayWenderlich Style Guide](https://github.com/raywenderlich/swift-style-guide)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

7. Architecture / Design Pattern

    - [ ] MVC  
    - [ ] MVP 
    - [ ] MVVM 
    - [ ] VIPER 

8. Architecture Implementation

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

9. SOLID

- Single Responsibility Principle 

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

- Open Closed Principle

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

- Liskov substitutability principle

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

- Interface segregation

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

- Dependency Inversion principle

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

10. Testing

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

11. Force unwrapping !

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

12. Mutable vs immutable property (Right use of `let`  and `var`)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

13. Retain cycles (Capture list)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

14. Code Documentation

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

### CODE REVIEW

File: TheMovieDb/Common/Utils/Extensions/UIViewControllerExtension.swift [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    }
+    
+    func removeMessageOnViewLabel() {
+        vMessageLabel?.removeFromSuperview()
+    }
+    
+    func showSpinner(onView: UIView) {
+        let spinnerView = UIView.init(frame: onView.bounds)
+        spinnerView.backgroundColor = .systemBackground
+        let ai = UIActivityIndicatorView.init(style: .large)
```
Comment (Kros-s) [inline TheMovieDb/Common/Utils/Extensions/UIViewControllerExtension.swift]: `ai` don't give clarity to the context, try to use a variable with a more meaningful name instead

File: TheMovieDb/Common/Utils/Extensions/UIViewControllerExtension.swift [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        ai.startAnimating()
+        ai.center = spinnerView.center
+        ai.color = UIColor(named: "BlueAppColor")
```
Comment (Kros-s) [inline TheMovieDb/Common/Utils/Extensions/UIViewControllerExtension.swift]: you can use a constant here to avoid mistakes

**Review by Jerriodas — COMMENTED** [review]
solve Kross comments and also and constants enum for magic numbers in your ViewControllers. Great Work!!

File: TheMovieDb/Common/Utils/Extensions/UIViewControllerExtension.swift (hunk 1 above)
Comment (javierCueto) [reply TheMovieDb/Common/Utils/Extensions/UIViewControllerExtension.swift]: fixed with a protocol extension and meaningful name

File: TheMovieDb/Common/Utils/Extensions/UIViewControllerExtension.swift (hunk 2 above)
Comment (javierCueto) [reply TheMovieDb/Common/Utils/Extensions/UIViewControllerExtension.swift]: done

**Comment by javierCueto** [comment]
magic numbers fixed :)

**Pull request merged**